app/models/coupon.rb

```diff
@@ -130,38 +131,23 @@ def can_edit?
     self.redemptions.empty?
   end
 
-  def validate_new_offers(offer_codes)
-    offer_codes = offer_codes.select(&:present?)
-    if offer_codes.blank?
-      self.errors.add(:offer, "code is required")
-    end
-    offer_codes.map do |offer_code|
-      offer = Offer.new(code: offer_code)
-      if !offer.valid?
-        self.errors.add(:offer, "code #{offer_code} is not valid")
-      end
-      offer
+  def update_coupon_offers(coupon, offer_codes)
+    Coupon.transaction do
+      self.update_offers(offer_codes)
+      self.update(coupon)
     end
```

Comment:
Added this to ensure that offer deletions are rolled back if needed, for example when prior offer codes are removed but the coupon updates are not saved.